node/setup.go

```diff
@@ -436,7 +444,9 @@ func createSwitch(config *cfg.Config,
 		p2p.SwitchPeerFilters(peerFilters...),
 	)
 	sw.SetLogger(p2pLogger)
-	sw.AddReactor("MEMPOOL", mempoolReactor)
+	if config.Mempool.Type != cfg.MempoolTypeNop {
+		sw.AddReactor("MEMPOOL", mempoolReactor)
```

Comment:
Will this also prevent declaring the mempool p2p channel(s)?

Reply:
Yes! Any nodes attempting to send us messages via mempool channels (meaning they have a flood mempool by mistake!) will be disconnected.

Reply:
This means that nodes configured by mistake with the flood mempool will not be able to communicate other kind of messages (consensus, sync, etc.) with this node. (I'm ok with that.)

Reply:
@hvanz I may be wrong, but last time I looked into it, upon handshake, the nodes will transmit/receive only on the intersection of their two channel lists

Reply:
> Any nodes attempting to send us messages via mempool channels (meaning they have a flood mempool by mistake!) will be disconnected.

_FYI: this statement was checked by running two nodes (one w/ flood mempool and another w/ nop mempool)_